runtime-modules/working-group/src/lib.rs

```diff
@@ -767,7 +941,23 @@ decl_module! {
 
         /// Sets a new status text for the working group.
         /// Requires root origin.
-        #[weight = 10_000_000] // TODO: adjust weight
+        ///
+        /// # <weight>
+        ///
+        /// ## Weight
+        /// `O (S)` where:
+        /// - `S` is the length of the contents of `status_text` when it is not none
+        ///
+        /// - DB:
+        ///    - O(1) doesn't depend on the state or parameters
+        /// # </weight>
+        #[weight = WeightInfoWorkingGroup::<T, I>::set_status_text(
+            if let Some(status_text) = status_text {
```

Comment:
Consider changing this snippet to a functional style: using Option.map with unwrap_or_default().